### PR TITLE
Use a random oracle, per the 2HashDH construction.

### PR DESCRIPTION
**Pull request by chris-wood**
As discussed!

cc @armfazh

### CODE REVIEW

File: draft-irtf-cfrg-voprf.md [hunk 1]
```diff
@@ -1043,43 +1043,54 @@ Steps:
...
+that provide strictly greater than 128 bits of security {{NIST}}.
 
 ## OPRF-curve448-HKDF-SHA512-ELL2-RO:
 
 - GG: curve448 {{RFC7748}}
   - HashToGroup(): curve448-SHA512-ELL2-RO
     {{I-D.irtf-cfrg-hash-to-curve}}
     - hash-to-curve DST: "RFCXXXX-OPRF-curve448-SHA512-ELL2-RO-"
-- H_1: HMAC_SHA512 {{RFC2104}}
+  - Serialize: The standard 56-byte representation of the public key
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-voprf.md:1063]: Typo :)

File: draft-irtf-cfrg-voprf.md [hunk 2]
```diff
@@ -1043,43 +1043,54 @@ Steps:
...
+- H_1: SHA512 {{RFC2104}}
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-voprf.md:1064]: We use a prefix-free encoding, which is sufficient to make it a RO per [https://link.springer.com/content/pdf/10.1007/11535218_26.pdf

File: draft-irtf-cfrg-voprf.md [hunk 3]
```diff
@@ -1043,43 +1043,54 @@ Steps:
...
 
 ## OPRF-P384-HKDF-SHA512-SSWU-RO:
 
 - GG: secp384r1 {{SEC2}}
   - HashToGroup(): P384-SHA512-SSWU-RO {{I-D.irtf-cfrg-hash-to-curve}}
     - hash-to-curve DST: "RFCXXXX-OPRF-P384-SHA512-SSWU-RO-"
-- H_1: HMAC_SHA512 {{RFC2104}}
+  - Serialize: A single byte set to 4, followed by the X-coordinate and
+    the Y-coordinate of the point, encoded as 32-byte big-endian
+    integers
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-voprf.md:1073]: We need to make a decision about compressed or not: I went with not compressed.

[the remark this answers is not in the document]
Comment (alxdavids) [reply draft-irtf-cfrg-voprf.md:1073]: Okay let's keep this for now and I'll create an issue to figure this out.

[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-voprf.md:1073]: Sounds good!

File: draft-irtf-cfrg-voprf.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-voprf.md:1063]: Hmm, I was basing this on RFC7748, which has 56 byte keys. (We're not using the Edwards form here.)

**Pull request merged**